lms/envs/common.py

```diff
@@ -3213,6 +3213,9 @@ def _make_locale_paths(settings):  # pylint: disable=missing-function-docstring
     'rules.apps.AutodiscoverRulesConfig',
     'bridgekeeper',
 
+    # management of user-triggered async tasks (course import/export, etc.)
+    'user_tasks',
+
```

Comment:
Can we take this out, since import/export is still only CMS?

Reply:
Tried to, haven't broken anything yet...

Reply:
Ah -- the import content from blockstore task is being run from xmodule code, which complains about the LMS not having user tasks...

>  File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/course_groups/cohorts.py", line 21, in <module>
>     from lms.djangoapps.courseware import courses
>   File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/courses.py", line 52, in <module>
>     from lms.djangoapps.courseware.block_render import get_block
>   File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/block_render.py", line 49, in <module>
>     from xmodule.library_tools import LibraryToolsService
>   File "/edx/app/edxapp/edx-platform/xmodule/library_tools.py", line 19, in <module>
>     from user_tasks.models import UserTaskStatus
>   File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/user_tasks/models.py", line 31, in <module>
>     class UserTaskStatus(TimeStampedModel):
>   File "/edx/app/edxapp/venvs/edxapp/lib/python3.8/site-packages/django/db/models/base.py", line 113, in __new__
>     raise RuntimeError(
> RuntimeError: Model class user_tasks.models.UserTaskStatus doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.

This is not something I am familiar with

Reply:
maybe this is unique to my local environment? I'm running my tests from the cms shell.

Reply:
Ah. That kinda makes sense. The task's top-level folder is part of what determines how the task is excuted. xmodule has never had celery tasks, so this is new territory. I'm a little fuzzy on the details but I think the platform is defaulting it to LMS workers.

It might be safer to put this task under cms/djangoapps/contentstore, along with all the other Studio tasks. If that'd cause an import cycle, then openedx/core/djangoapps/content_libraries is another place you could put it, although I think there is some decorator you'd need to add to make it run only in CMS.

Reply:
The only refactoring which is feasible is replacing the user tasks model with a bespoke one, which feels like a waste of energy.

Reply:
https://github.com/openedx/edx-platform/blob/db459761d86ae1379344d6fcb8eda048674b8f80/.github/workflows/unit-test-shards.json#L157 THis might help me understand how tests are run & things are split up? Which doesn't solve this problem but might give some insight.